### DISCUSSION TITLE
Exit code: 1 nd</callCatchAppNotFound/

### DISCUSSION THREAD

**Discussion by zepinos** [aclap-dev/video-downloadhelper#377] [Help and Q&A] [answered]
### Your question

Hello and greeting.

I would like to seek an advice or solution regarding my issue.
All this while I was able to download videos from website. However, when I tried for the first time at "inflearn.com" site below, I was unable to download it. 

Exit code: 1

moz-extension://d5ac8cb1-ec18-480a-9632-337354a2db0b/background/main.js:1
receive@moz-extension://d5ac8cb1-ec18-480a-9632-337354a2db0b/background/main.js:1:13157
nd</callCatchAppNotFound/</<@moz-extension://d5ac8cb1-ec18-480a-9632-337354a2db0b/background/main.js:1:42624

pageUrl: https://www.inflearn.com/course/lecture?courseSlug=%EC%95%A0%EC%A0%80-%ED%81%B4%EB%9D%BC%EC%9A%B0%EB%93%9C-%EB%B3%B4%EC%95%88&unitId=117134

core_media:
```
{
"builder":"NakedHls"
"protocol":"hls"
"duration":328.999987
"container":{
"name":"Mp4"
"extension":"mp4"
"audio_only_extension":"mp3"
"defacto_codecs":{
"audio":{
}
"video":{
}
}
"supported_video_codecs":[
0:
"H264"
1:
"H265"
2:
"Av1"
3:
"MP4V"
4:
"MPEG2"
5:
"unknown"
]
"supported_audio_codecs":[
0:
"Opus"
1:
"MP3"
2:
"FLAC"
3:
"AAC"
4:
"unknown"
]
"mimetype":/(?:x-)?mp4/i
}
"av":{
"audio":false
"video":{
"codec":{
"name":"H264"
"type":"video"
"mimetype":/avc1.*/i
"defacto_container":"Mp4"
}
"fps":{
}
"dimensions":{
}
"quality":{
}
"bitrate":{
}
}
}
}
```
headers:
```
[
0:{
"name":"User-Agent"
"value":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/201001 ..."
}
1:{
"name":"Accept"
"value":"*/*"
}
2:{
"name":"Accept-Language"
"value":"ko-KR,ko;q=0.8,en-US;q=0.5,en;q=0.3"
}
3:{
"name":"Accept-Encoding"
"value":"gzip, deflate, br"
}
4:{
"name":"Referer"
"value":"https://www.inflearn.com/"
}
5:{
"name":"Origin"
"value":"https://www.inflearn.com"
}
6:{
"name":"DNT"
"value":"1"
}
7:{
"name":"Connection"
"value":"keep-alive"
}
8:{
"name":"Cookie"
"value":"CloudFront-Policy=eyJTdGF0ZW1lbnQiOlt7IlNpZCI6MTM3MDQ3NywiUmVzb3VyY2UiOiJodHRwczovL3ZvZC5pbmZsZWFybi5jb20vdmlkZW9zLzM1NzAyY2U5LWY2OWYtNDkyNS05NWY3LWMxOWQxM2ZmN2YzYS9lbmNyeXB0ZWQvKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTcxMjI5MjQ1Nn19fV19; CloudFront-Key-Pair-Id=APKAITJXMBGW53HFS7RQ; CloudFront-Signature=AGpHC1rXRcyUu4IFImMya5Mzc-U19PjvNuEIXHAucPNWACMVAvkTMqE-08i9Lii3ukcocmjfNFWI1Dr5aLQshovHdEXIDVtrAXQhioAy1q4i7cgmqViUOypkM7MwnbqMRKFiFpGwT3De~5p8IvG9vENvsQgINzqd0bPHir30mUj64nj5IQwwPH0rQH7lJqJ8mJQmjvoATHxIA9ekT-N9DrRPkPD~IU6FVqjNJl0yvuJERDBLHbz3qEOt8UkYendraYE~yNY5ClKQ-DpHlEOzIpIrAHNOFdkZOnlyCXahODUIXDbZcXXHE6pBtWrQTrr56LoPLyBqi2kr3XzUP9iBcQ__; deviceId=a4f26a45-0c57-4202-81ec-a8a40cbd320a; _hackle_hid=97453155-a67e-447f-9bc1-79d7f896c436; _hackle_session_id_39swIGfTAhDcz0QXvmh6hayS=1712281649777.67da81c7; _hackle_last_event_ts_39swIGfTAhDcz0QXvmh6hayS=1712281650131; _hackle_mkt_rb1wDPCB=%7B%7D; ch-veil-id=2ee897b1-28a7-4538-93e6-4579f2600eaf; connect.sid=s%3AFbc-oH-H3tyhUaTN-7OoZKtmVQP1fyFt.o6CnFR%2FbvVMSoGx5rwHfOTgrcjxBH1hcbaLKKqQ7lfU; _dd_s=rum=1&id=27ebee9b-f863-445a-95d9-645d4db52e04&created=1712275626891&expire=1712282549746; _gcl_au=1.1.187298227.1712278054; _ga_85V6SRKGJV=GS1.1.1712278054.1.1.1712278378.60.0.0; _ga=GA1.1.895537797.1712278055; _fwb=312F8IKtD6SI1YLw1OXo4y.1712278054536; AUFAH4A45343888168=1712278059802864406|2|1712278059802864406|1|1712278059803164406; _fbp=fb.1.1712278055851.353836823; _hjSessionUser_1039300=eyJpZCI6IjY2ZmJiMWIwLTMzMzctNTkzYy04NTUzLTUxZTliYzM2MmRmNSIsImNyZWF0ZWQiOjE3MTIyNzgwNTU4NzgsImV4aXN0aW5nIjpmYWxzZX0=; _hackle_uid_rb1wDPCB39swIGfTAhDcz0QXvmh6hayS=1370477"
}
9:{
"name":"Sec-Fetch-Dest"
"value":"empty"
}
10:{
"name":"Sec-Fetch-Mode"
"value":"cors"
}
11:{
"name":"Sec-Fetch-Site"
"value":"same-site"
}
]
```

works fine on other sites.
Thank you in advanced. Your help is greatly appreciated.

### Addon & CoApp version and details

version: 8.2.2.8a1
channel: beta
build date: 2024-03-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: ko
license: locked
key (only 16 first characters): ca70090adfbd4797
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\~~~",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by Mattmjs** [comment]
It is another site using a non standard way of streaming the media, there was another case matching this one in this discussion.
[https://github.com/aclap-dev/video-downloadhelper/discussions/162](url)

It was suggested to try an older of VDH, 1.6.3 companion app. An older extension might be needed also
https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/

That would be quite a downgrade though from the current software.

**Reply by zepinos** [reply]
https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/

I installed and checked all versions of the link above. Version 7.0.0 stopped and did not proceed, and all other versions encountered errors.

Among the old versions, there is a log that allows you to understand the contents more.

```
[hls @ 00000116db2fadc0] Unable to open key file https://vod.inflearn.com/key/35702ce9-f69f-4925-95f7-c19d13ff7f3a/0
[hls @ 00000116db2fadc0] Unable to open key file https://vod.inflearn.com/key/35702ce9-f69f-4925-95f7-c19d13ff7f3a/1
[hls @ 00000116db2fadc0] Unable to open key file https://vod.inflearn.com/key/35702ce9-f69f-4925-95f7-c19d13ff7f3a/2
[hls @ 00000116db2fadc0] Unable to open key file https://vod.inflearn.com/key/35702ce9-f69f-4925-95f7-c19d13ff7f3a/3
[hls @ 00000116db2fadc0] Error when loading first segment 'https://vod.inflearn.com/videos/35702ce9-f69f-4925-95f7-c19d13ff7f3a/encrypted/1_0.ts'
[in#0 @ 00000116d99a9ec0] Error opening input: Invalid data found when processing input
Error opening input file https://vod.inflearn.com/videos/35702ce9-f69f-4925-95f7-c19d13ff7f3a/encrypted/1.m3u8.
Error opening input files: Invalid data found when processing input
```

The site is www.inflearn.com , but the file is calling vod.inflearn.com . When the developer tool changed the www.inflearn.com request to the vod.inflearn.com address and called it, it was 200 instead of a 403 error.

I don't know if it's a cookie or something, but can't VDH handle this part?

I apologize in advance for my inexperience in English.
Thank you in advanced. Your help is greatly appreciated.

**Reply by Mattmjs** [reply]
I tried extension version 8.0.0.6 and companion 1.6.3 in Linux on a virtual machine. All I got was **Response code 403 (Forbidden)** and **Aborted** 

Sorry I can't assist you more on this, I'm just a user like you. The site does not make it easy for downloading because of the non standard method of streaming. It would be easier to screen record this. Probably not what you want to do but it'll work.

**Reply by zepinos** [reply]
Your answer helped me a lot.

From what I've experimented with, since it is a site that requires authentication, the result is returned with a 403 Forbidden by default. If I change the address to the developer tool while authenticating with www and send a request again, it will return to 200 OK, so it seems that there is something that is done with the HTTP request header.

If there is anyone who can analyze or improve the program, it would be better to provide site authentication information, but since it is a site in Korean, it will be difficult to analyze due to language problems. Is there any other good way?

I'm not considering using other programs that record the screen. There are too many things to record to use that method.

Once again, thank you for your help, @Mattmjs.

**Reply by Mattmjs** [reply, marked as answer]
I'll give you a response by GrampaWildWilly on the same discussion I linked to above, who figured out how to get the video.
If you can understand what he's talking about, it requires a bit of manual work. https://github.com/aclap-dev/video-downloadhelper/discussions/162#discussioncomment-8541847
